### DISCUSSION TITLE
New user datatype - how to use?

### DISCUSSION THREAD

**Discussion by andyburgessmd** [Budibase/budibase#11986] [Help]
I am trying to update my application to use the new user type instead of a relationship to the user table but I can't figure out how to use it in a filter on a table.
Previously I could use `Current User.table` but now I need to just use `table `and add a filter where User equals a binding of something in Current User object. I've tried all the values (_id, email, etc) but none seem to match?

**Comment by andyburgessmd** [comment]
Another example is if you add a `User Field` to a form and want to set its default value to current user. What part of the `Current User` object to you use?

**Comment by ammo8600** [comment]
Same here, can't seem to find a way to bind to the user column while filtering.

**Comment by mike12345567** [comment]
Hi guys - sorry there was a slight issue which has now been corrected with the filtering, if you're on self host make sure to update to `2.11.14+`.

For filtering, you need to search by the `{{ Current User._id }}` field - we utilise the user ID to lookup as this is something that does not change. If you want to lookup a different user, then you can filter the user table by, email, first/last name etc and take the user ID from the results to search other rows by.

**Comment by andyburgessmd** [comment]
Thanks @mike12345567 that works for the filter.
But I'm still having issues with the User Field component in a form. The value shows as null if I try to set a default value to the current user or in an update form to the value coming from the dataprovider/repeater.

**Reply by mike12345567** [reply]
Hi @andyburgessmd - i've made a small backend adjustment so that the `{{ Current User._id }}` can be saved - theres a small visual bug where it shows as `null` in the dropdown if its set as a default value, but that will be fixed soon.

**Reply by andyburgessmd** [reply]
How long will the front end fix take?
Also to add to it the User component does not seem to list or allow search of all the users in an application? In this example I can't find myself.
![image](https://github.com/Budibase/budibase/assets/110813989/ef29e6ed-051f-4d6e-b2ae-a45db285b5ec)
Note: Neither of the two redacted users is me.

**Reply by mike12345567** [reply]
Hi @andyburgessmd - the fixes should be going out today - there is a set of improvements that will be coming with it. We are also working on a feature to help with migration of the the old user relationship type to the new user column type, but this will take a bit longer to complete.

**Reply by andyburgessmd** [reply]
@mike12345567 I've updated to 2.11.30 today and  the issues I have mentioned are still open. From initial checks I can still see
- User field in a form still shows null instead of a current user email address - this happens in both a create form and update with data provider
- User field in a form still does not show all users that are in the users table

**Reply by mike12345567** [reply]
Hi @andyburgessmd - could you tell me how you've configured the form to create that scenario? If I use a default value like `{{ Current User._id }}` on 2.11.30 this appears to be resolved (you may need to update your client app version).

As for the user field in a form, is there anyway you can give me a bit more info about this? I'm not able to replicate, the form should show all of the users in your Budibase installation, if you go into the portal it should present all of the users shown in the users table (you may need to search for some users as it won't list every user if you have a lot of them).

EDIT: I believe I may have found a way to replicate it - I will get a fix together quickly!

**Reply by andyburgessmd** [reply]
Thanks @mike12345567. I was just pulling together screenshots from my application for you based on the other bit of your reply. Do you still need those?

**Reply by mike12345567** [reply]
Hi @andyburgessmd - no I think I've worked it out, will message here when fix is available.

**Reply by andyburgessmd** [reply]
@mike12345567 haven't heard back yet but have been testing on 2.11.35 and user field is still showing as null in an update form when using the data from the data provider, and although the form now shows all users it also shows users that are not users of the application.

**Reply by mike12345567** [reply]
Hi @andyburgessmd - how have you configured the update form to get it to show null? As for the user list - today we've decided to show all users within the Budibase environment rather than those assigned to the app - but definitely could be a good option to add to allow filtering it down to just the users that have a role within the app if that is desired - feel free to open an enhancement issue if you'd like to see an option like this!

**Reply by andyburgessmd** [reply]
@mike12345567 The update form is configured with these settings (hopefully all that are relevant) and it still shows null.
![image](https://github.com/Budibase/budibase/assets/110813989/945c5dba-6792-4eaf-b078-34e19042d859)
![image](https://github.com/Budibase/budibase/assets/110813989/217159ec-1b64-422f-b233-0ab80e5e7f25)
![image](https://github.com/Budibase/budibase/assets/110813989/3c8b9d36-6655-48b8-9a5c-18bc50dc66df)

I will send an enhancement for that feature.

But have also just raised another bug related to the users - https://github.com/Budibase/budibase/issues/12088

**Comment by andyburgessmd** [comment]
Enhancement request for filtering users is https://github.com/Budibase/budibase/issues/12089 if anyone wants to like or follow.
